src/migrations/m180103_040100_log_change_level_to_int.php

```diff
@@ -1,74 +0,0 @@
-<?php
-/**
- * @link http://www.yiiframework.com/
- * @copyright Copyright (c) 2008 Yii Software LLC
- * @license http://www.yiiframework.com/license/
- */
-
-use yii\base\InvalidConfigException;
-use yii\db\Migration;
-use yii\log\DbTarget;
-
-/**
- *
- * @author Dzhuneyt Ahmed <dzhuneyt@dzhuneyt.com>
- */
-class m180103_040100_log_change_level_to_int extends Migration
-{
-
-    /**
-     * @var DbTarget[] Targets to create log table for
-     */
-    protected $dbTargets = [];
-
-    /**
-     * @throws InvalidConfigException
-     * @return DbTarget[]
-     */
-    protected function getDbTargets()
-    {
-        if ($this->dbTargets === []) {
-            $logger = Yii::getLogger();
-            if (!$logger instanceof \yii\log\Logger) {
-                throw new InvalidConfigException('You should configure "logger" to be instance of "\yii\log\Logger" before executing this migration.');
-            }
-
-            $usedTargets = [];
-            foreach ($logger->targets as $target) {
-                if ($target instanceof DbTarget) {
-                    $currentTarget = [
-                        $target->db,
-                        $target->logTable,
-                    ];
-                    if (!in_array($currentTarget, $usedTargets, true)) {
-                        // do not create same table twice
-                        $usedTargets[] = $currentTarget;
-                        $this->dbTargets[] = $target;
-                    }
-                }
-            }
-
-            if ($this->dbTargets === []) {
-                throw new InvalidConfigException('You should configure "log" component to use one or more database targets before executing this migration.');
-            }
-        }
-
-        return $this->dbTargets;
-    }
-
-    public function up()
-    {
-        foreach ($this->getDbTargets() as $target) {
-            $this->db = $target->db;
-            $this->alterColumn($target->logTable, 'level', $this->integer());
-        }
-    }
-
-    public function down()
-    {
-        foreach ($this->getDbTargets() as $target) {
-            $this->db = $target->db;
-            $this->alterColumn($target->logTable, 'level', $this->string());
-        }
-    }
-}
```

Comment:
Usually it's almost always a bad idea to delete a migration. Instead, a new migration reverting the change should be added. 

But since Yii3 isn't out yet, I'm not sure. @samdark ?

Reply:
> But since Yii3 isn't out yet

It does not matter - there should be upgrade path from Yii2 to Yii3, so migration should stay as is. 

But honestly I doubt that these migrations should exist in the first place. It is ugly (it depends on app state, co it will work differently depending on time when it is run) and impractical (how can I create logs tables when I add additional targets after some time?). I would leave these migrations as is and create some command that will create and update tables for all available targets, so you could run it anytime after adding new target.

Reply:
I think that we can safely delete that migration, because it is in tests folder. So I don't think that it will be called when log package is installed.

Reply:
Create another one.

Reply:
This migration can be removed - it was never released, so there is no need to keep it (especially if it will change type from `int` to `int`... :D).

Reply:
Sounds safe enough to just remove it if nobody ever used it, yes :)

Reply:
Ok, so I'll leave so